pkg/validation/limits.go

```diff
@@ -650,6 +652,11 @@ func (o *Overrides) BlockedQueries(ctx context.Context, userID string) []*valida
 	return o.getOverridesForUser(userID).BlockedQueries
 }
 
+func (o *Overrides) RequiredLabels(ctx context.Context, userID string) []string {
+	// todo (callum), this probably needs to take a list of the labels present?
```

Comment:
You need to join this list with the one from the context.

Reply:
Yes, that would be done in the QueryLimits Limiter when we add this `RequiredLabels` limit (follow up PR)